src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -249,4 +271,19 @@ private void writeCertificateAndKeyToPath(CreateKeysAndCertificateResponse respo
             throw new RuntimeException(e);
         }
     }
+
+    private void writeCertificateToPath(CreateCertificateFromCsrResponse response, String rootPath) {
+        try {
+            Path certPath = Paths.get(rootPath, DEVICE_CERTIFICATE_PATH_RELATIVE_TO_ROOT);
+            if (Files.notExists(certPath)) {
+                Files.createFile(certPath);
+            }
```

Comment:
these lines aren't needed, just using Files.write should be sufficient to create the file with contents if it doesn't already exist.

Reply:
fixed